test/manual/kerberos.test.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+const { MongoClient } = require('../../src');
+const chai = require('chai');
+const expect = chai.expect;
+
+describe('Kerberos', function () {
+  if (process.env.MONGODB_URI == null) {
+    console.log('skipping Kerberos tests, MONGODB_URI environment variable is not defined');
+    return;
+  }
+  let krb5Uri = process.env.MONGODB_URI;
+  if (process.platform === 'win32') {
+    console.log('Win32 run detected');
+    if (process.env.LDAPTEST_PASSWORD == null) {
+      throw new Error('The env parameter LDAPTEST_PASSWORD must be set');
+    }
+    const parts = krb5Uri.split('@', 2);
+    krb5Uri = `${parts[0]}:${process.env.LDAPTEST_PASSWORD}@${parts[1]}`;
+  }
+
+  it('should authenticate with original uri', function (done) {
+    const client = new MongoClient(krb5Uri);
+    client.connect(function (err, client) {
+      expect(err).to.not.exist;
+      client
+        .db('kerberos')
+        .collection('test')
+        .find()
+        .toArray(function (err, docs) {
+          expect(err).to.not.exist;
+          expect(docs[0].kerberos).to.be.true;
+
+          client.close(done);
+        });
+    });
+  });
+
+  it('validate that SERVICE_REALM and CANONICALIZE_HOST_NAME can be passed in', function (done) {
+    const client = new MongoClient(
+      `${krb5Uri}&authMechanismProperties=SERVICE_NAME:mongodb,CANONICALIZE_HOST_NAME:false,SERVICE_REALM:windows&maxPoolSize=1`
+    );
+    client.connect(function (err, client) {
+      expect(err).to.not.exist;
+      client
+        .db('kerberos')
+        .collection('test')
+        .find()
+        .toArray(function (err, docs) {
+          expect(err).to.not.exist;
+          expect(docs[0].kerberos).to.be.true;
+
+          client.close(done);
+        });
+    });
+  });
+
+  it('should authenticate with additional authentication properties', function (done) {
+    const client = new MongoClient(
+      `${krb5Uri}&authMechanismProperties=SERVICE_NAME:mongodb,CANONICALIZE_HOST_NAME:false&maxPoolSize=1`
+    );
+    client.connect(function (err, client) {
+      expect(err).to.not.exist;
+      client
+        .db('kerberos')
+        .collection('test')
+        .find()
+        .toArray(function (err, docs) {
+          expect(err).to.not.exist;
+          expect(docs[0].kerberos).to.be.true;
+
+          client.close(done);
+        });
+    });
+  });
+
+  it.skip('should fail to authenticate due to illegal service name', function (done) {
```

Comment:
Note - I had to skip this test because it connects successfully. This makes me wonder if the above tests which add additional query parameters are actually doing anything, but I left them in there since they pass.

Reply:
Yeah here's [an example log](https://spruce.mongodb.com/task/mongo_node_driver_ubuntu_14.04_dubnium_test_auth_kerberos_patch_4fcb031bb753facbc6d9a4ebffe6279a3dd81941_5f2ab83461837d0cc464af1f_20_08_05_13_46_49/logs). The `gssapiServiceName` doesn't seem to matter; maybe it gets ignored?